src/ShopBundle/objects/WebModules/ViewMyOrderDetails/MTShopViewMyOrderDetails.php

```diff
@@ -77,29 +61,28 @@ public function Accept(
         $orderIdRequested = $oVisitor->GetSourceObject('orderIdRequested');
         $extranetUserId = $oVisitor->GetSourceObject('extranetUserId');
 
+        $order = $this->getDbAdapter()->getOrder($orderIdRequested);
+
         $viewData = array(
             'error' => false,
             'errorCode' => null,
             'notMyOrderError' => false,
             'orderNotFoundError' => false,
-            'order' => null,
+            'order' => $order,
         );
 
-        if (false === $viewOrderDetailHandler->orderIdBelongsToUser($orderIdRequested, $extranetUserId)) {
-            // this is not our order. access denied
-            $viewData['error'] = true;
-            $viewData['errorCode'] = 'notMyOrderError';
-        }
-
-        $order = $this->getDbAdapter()->getOrder($orderIdRequested);
         if (null === $order) {
             // this is not our order. access denied
```

Comment:
Comment is slightly off - as it is the same as below but a different case.

Reply:
Removed the comments as the code is explicit enough.